core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -82,6 +76,46 @@ def start_navigator(self) -> None:
     def start_serial(self, device: str) -> None:
         self.start_mavlink_manager(Endpoint("Master", self.settings.app_name, "serial", device, 115200, protected=True))
 
+    @staticmethod
+    def get_current_arch() -> str:
+        """Maps paltform.machine() outputs to docker architectures"""
+        arch_map = {"x86_64": "amd64", "armv7l": "arm"}  # TODO: differentiate arm64v8/ arm32v7
+        local_arch = platform.machine()
+        arch = arch_map.get(local_arch, None)
+        if not arch:
+            raise Exception(f"Unknown architecture ('{local_arch}') is not supported.")
+        return arch
+
+    def run_with_sitl(self) -> None:
```

Comment:
run_with_sitl should receive an argument of vehicle type.

Reply:
Done.